intensityAnalysis.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+
+"""
+Created on Tue Aug 20 10:56:31 2024
+
+@author: LawineNaturgefahren
+"""
+
+'''
+Intensity characteristics AMA thalwege
+'''
+
+import pathlib
+import numpy as np
+from shapely import get_coordinates
+import math
+import avaframe.in3Utils.geoTrans as gT
+import avaframe.in2Trans.ascUtils as IOf
+
+def intensityCharacteristics (dbData, resDist):
+    
+    for index, row in dbData.iterrows():
+        
+        # total length of thalweg
+        x = get_coordinates(row['geom_path_ln3d_epsg:31287_resampled'])[:,0]
+        y = get_coordinates(row['geom_path_ln3d_epsg:31287_resampled'])[:,1]
+        
+    
+        
+        # fetch DEM
+        demPath = pathlib.Path('data', 'amaExports', row['path_name'], ('dem_path_%d.asc' % row['path_id']))
+        dem = IOf.readRaster(demPath)
+    
+        # setup avaPath to get parabolic fit
+        avaPath = {'x': x, 'y': y}
+        # first resample path and make smoother - TODO check if required - double resampling check line 56
+        avaPath, projPoint = gT.prepareLine(dem, avaPath, distance=resDist, Point=None)
+        
+    
+        opoint =  {'x':row['geom_origin_pt3d_epsg:31287_snapped'].xy[0], 'y':row['geom_origin_pt3d_epsg:31287_snapped'].xy[1]}
+        origin = gT.findClosestPoint(avaPath['x'], avaPath['y'], opoint)
+        
+        tpoint =  {'x':row['geom_transit_pt3d_epsg:31287_snapped'].xy[0], 'y':row['geom_transit_pt3d_epsg:31287_snapped'].xy[1]}
+        transit = gT.findClosestPoint(avaPath['x'], avaPath['y'], tpoint)
+        
+        dpoint = {'x':row['geom_runout_pt3d_epsg:31287_snapped'].xy[0], 'y':row['geom_runout_pt3d_epsg:31287_snapped'].xy[1]}
+        depo = gT.findClosestPoint(avaPath['x'], avaPath['y'], dpoint)
+        
+        dbData.at[index,'depoID'] = depo
+        dbData.at[index,'transitID'] = transit
+        dbData.at[index,'origID'] = origin
```

Comment:
all of this could be moved to a separate function - finding indices of origin, deposition, transit points - as this is also used in the fit functions for example